crates/unftp-auth-jsonfile/src/lib.rs

```diff
@@ -210,7 +212,40 @@ struct UserCreds {
 impl JsonFileAuthenticator {
     /// Initialize a new [`JsonFileAuthenticator`] from file.
     pub fn from_file<P: AsRef<Path>>(filename: P) -> Result<Self, Box<dyn std::error::Error>> {
-        let json: String = fs::read_to_string(filename)?;
+        let mut f = fs::File::open(&filename)?;
+
+        // The credentials file can be plaintext, gzipped, or gzipped+base64-encoded
+        // The gzip-base64 format is useful for overcoming configmap size limits in Kubernetes
+        let mut magic: [u8; 4] = [0; 4];
+        let n = f.read(&mut magic[..])?;
+        let is_gz = n > 2 && magic[0] == 0x1F && magic[1] == 0x8B && magic[2] == 0x8;
+        // the 3 magic bytes translate to "H4sI" in base64
+        let is_base64gz = n > 2 && magic[0] == b'H' && magic[1] == b'4' && magic[2] == b's' && magic[3] == b'I';
+
+        f.rewind()?;
+        let json: String = match is_gz | is_base64gz {
+            true => {
```

Comment:
Since you are only comparing to true and false you might as well use an `if .. else` expression. You can return values from it like you do with the match.

Reply:
Right, I did it again 🤦‍♂️

This PR was already, I made a new PR for this fix. See #443